koap/src/main/kotlin/Decoder.kt

```diff
@@ -206,13 +206,13 @@ fun ByteArray.decodeTcp(): Message.Tcp {
     // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
     // |  Options (if any) ...
     // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    val options = content.readOptions()
+    val options = if (!content.exhausted()) content.readOptions() else emptyList()
 
     // |7 6 5 4 3 2 1 0|7 6 5 4 3 2 1 0|7 6 5 4 3 2 1 0|
     // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
     // |    Payload (if any) ...
     // +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
-    val payload = content.readByteArray()
+    val payload = if (!content.exhausted()) content.readByteArray() else byteArrayOf()
```

Comment:
The `exhausted` check is not actually needed here, as `readByteArray` will return an empty `ByteArray` if there are no more bytes to read.

```suggestion
    val payload = content.readByteArray()
```